### ISSUE TITLE
Disk usage does not show its value when downloading a game

### ISSUE THREAD

**Issue by jul-ben** [ValveSoftware/steam-for-linux#7261]
#### Your system information

* Steam client version (build number or date): Jul 8 2020, 23:11:02
* Distribution (e.g. Ubuntu): Debian 10 (Buster)
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:

When downloading a game via the Steam client, I would expect to see the value of disk usage in the "Downloading" page.

#### Steps for reproducing this issue:

1. Open the Steam client
2. Install a game via the Steam client
3. Go to the downloads page by clicking on "DOWNLOADS" at the bottom of the client window
4. "Disk usage" value will stay at "0" throughout the whole download

#### Example (screenshot):

![Screenshot from 2020-07-13 04-52-03](https://user-images.githubusercontent.com/9112511/87266321-3da44a80-c4c5-11ea-8ae1-f9672e2e6e93.png)

**Comment by kortewegdevries** [comment]
Does this issue occur for this game only or on other games? And you can check if I/O is _actually_ being done by steam using `sudo iotop --user=Username --only` after starting a download; it won't show an idle steam doing I/O as far as I know. Either that or something is wrong because it shows fine on my Ubuntu laptop and Debian VM.

**Comment by jul-ben** [comment]
So I got some interesting results:

`iotop` always returns that there is writing on the SSD (the total disk write is consistent with the usage), regardless of the title considered.

For example:
![Screenshot from 2020-07-14 01-37-39](https://user-images.githubusercontent.com/9112511/87364081-f2e10c00-c572-11ea-9268-43f37e363485.png)

On the other hand, it seems to only occur with this title, which is not out yet. So, I am wondering if it was not happening only because this is a "preload" for a game that is not out yet. If this is true I would only be able to to do a couple of tests before this disappears.

1) (Another) Preload of Death Stranding:

![Screenshot from 2020-07-14 01-43-08](https://user-images.githubusercontent.com/9112511/87364269-86b2d800-c573-11ea-91d9-d5e1d3d9d427.png)

2) Download of The Surge 2:

![Screenshot from 2020-07-14 01-44-51](https://user-images.githubusercontent.com/9112511/87364326-b1049580-c573-11ea-98dc-6f8d7099889c.png)

You can even see the difference in the green graph (which displays disk usage over time).

**Comment by kortewegdevries** [comment]
https://partner.steamgames.com/doc/sdk/uploading#Preloading

**Comment by jul-ben** [comment]
> https://partner.steamgames.com/doc/sdk/uploading#Preloading
> 
> From what I understand it's downloading encrypted compressed chunks to your system, because it's isn't being read at the moment,the graph isn't showing a disk usage.

Nice finding.

Not sure whether you are saying "it is not a bug, it is a feature" or reading the disk usage does not work for encrypted data.

If it the latter, it is "just" lacking an implementation for it ; encrypting a content does not mean you cannot read the data amount being written on disk.

**Comment by kortewegdevries** [comment]
It launched today, did it show a disk usage? I'm not sure how they implement or calculate that graph, couldn't find anything on it.

**Comment by jul-ben** [comment]
> It launched today, did it show a disk usage? I'm not sure how they implement or calculate that graph, couldn't find anything on it.

Yes, like I any other released title. So, obviously, I cannot reproduce the issue anymore.

I do not own any other title which would be released soon.

**Comment by kortewegdevries** [comment]
> I do not own any other title which would be released soon.

I think it's perfectly normal for a preloading to show no disk usage; basing this on anecdotal evidence [1](https://steamcommunity.com/app/337000/discussions/0/355043117518541479/), [2](https://steamcommunity.com/discussions/forum/1/496880503062850443/),[3](https://www.reddit.com/r/pcmasterrace/comments/3s2pye/steam_preload_0_disk_usage/),[4](https://steamcommunity.com/app/377160/discussions/0/496880503081860193/) etc.

**Comment by jul-ben** [comment]
> I think it's perfectly normal for a preloading to show no disk usage; basing this on anecdotal evidence 

I strongly disagree: a lack of implementation of this feature does not mean "it is normal".

**Comment by kortewegdevries** [comment]
> I strongly disagree: a lack of implementation of this feature does not mean "it is normal".

Yeah, sorry I meant, "given the sources I could find I think it's normal to assume that the client might not have this 'feature' as of now hence this is an 'enhancement' and not a 'bug' as I thought at first "
